docs/understanding-dependencies.md

```diff
@@ -0,0 +1,24 @@
+# dependencies vs. devDependencies
+
+> What's the difference between these two properties in `package.json`?
+
+While functionally almost identical, there are small differences that set these two apart. At its core, both `dependencies` and `devDependences` properties in `package.json` are where external packages that your project uses are defined. Both properties contain an object of name/versions. Each time you `npm install` or `npm install --save-dev` a new entry is added to either `dependencies` or `devDependencies` respectively, in the format of 
+```json
+  "name": "version"
```

Comment:
Would it be useful to link to something explaining semantic versioning and how NPM uses `^`, `>=`, `~`, etc.? I know this was a big stumbling block for me when I was first learning NPM's ecosystem.

The obvious criticism to this is that it's too much, too fast, but I think it should at least be mentioned somewhere - I don't remember the discussion thread now, but someone mentioned informational popups after the user had used Guppy for a certain amount of time, as well as unlocking more advanced UI elements, I think this should be shared at some point before they're expected to read/edit the `package.json` themselves.

Reply:
I definitely think it would be worthwhile to include. @joshwcomeau thoughts? Also, we *could* generalize this doc into more of an 'about package.json' informational doc vs. 'deps vs devDeps' specifically?

Reply:
+1 that idea - just link directly to the deps/devDeps section from within the app, but have the rest of the material there to cover package.json/general NPM package structure.